scripts/validate_snapshots.py

```diff
@@ -0,0 +1,110 @@
+"""
+Command line utility to validate snapshots prior to indexing by Azul
+"""
+
+import argparse
+from collections import (
+    defaultdict,
+)
+import json
+import logging
+import sys
+
+from azul import (
+    config,
+)
+from azul.args import (
+    AzulArgumentHelpFormatter,
+)
+from azul.azulclient import (
+    AzulClient,
+)
+from azul.bigquery import (
+    backtick,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+from azul.terra import (
+    TDRSourceSpec,
+)
+
+log = logging.getLogger(__name__)
+configure_script_logging(log)
+
+
+def main(args):
+    joiner = '||'
+    deny_snapshots = defaultdict(dict)
+
+    azul = AzulClient(num_workers=1)
+    sources_by_catalog = {
+        catalog: azul.catalog_sources(catalog)
+        for catalog in args.catalogs
+    }
+    for catalog, sources in sources_by_catalog.items():
+        tdr_plugin = azul.repository_plugin(catalog)
+        log.info('Checking for %r in catalog %s', joiner, catalog)
+        if args.snapshot is not None:
+            sources = [s for s in sources if args.snapshot in s]
+            if len(sources) < 1:
+                continue
```

Comment:
```suggestion
```

Not needed. Generally we avoid bail-out statements, and with it removed the following `for` loop will just run zero times if `sources` is empty.